### PR TITLE
(HC-6) Port ConfigDocument

### PR DESCRIPTION
**Pull request by MikaelSmith**

### CODE REVIEW

File: lib/src/nodes/config_node_object.cc [hunk 1]
```diff
@@ -168,11 +168,11 @@ namespace hocon {
                 auto single = dynamic_pointer_cast<config_node_single_token>(before_last);
                 if (single && single->get_token()->get_token_type() == token_type::IGNORED_WHITESPACE) {
                     indent = single->get_token()->token_text();
-                    indent += "  ";
-                    indentation.push_back(make_shared<config_node_single_token>(
-                            make_shared<ignored_whitespace>(nullptr, indent)));
-                    return indentation;
                 }
+                indent += "  ";
```
Comment (MikaelSmith) [inline lib/src/nodes/config_node_object.cc]: @Magisus indentation was broken here.

Comment (Magisus) [inline lib/src/nodes/config_node_object.cc]: Sweet, thanks for finding that!

**Comment by MikaelSmith** [comment]
Should now pass the tests.

**Comment by MikaelSmith** [comment]
I think this is ready, I made a few changes and I'm happy with it.

**Comment by MikaelSmith** [comment]
This was really HC-9, the ConfigDocumentFactory. Oops.

**Pull request merged**